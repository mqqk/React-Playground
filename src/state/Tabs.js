import React from 'react';
// import ReactDOM from 'react-dom';


export default class Tabs extends React.Component{

    static defaultProps = {
        tabs:[]
    };

    state = {
        currentTabIndex:0,
    }

    handleButtonClick = (index) => { //still not sure how to talk through this line of code
        console.log('button clicked',{index})
        this.setState({currentTabIndex:index})
        
    }
//i don't know how to read line 23
    renderButtons(){
        return this.props.tabs.map((tab,index) => (
            <button key={index} onClick={() => this.handleButtonClick(index)}> 
                {tab.name}
            </button>
        ))
    }

    renderContent(){
        const currentTab = this.props.tabs[this.state.currentTabIndex];//i don't understand how the state was absorb in to the array of tabs
        return(
            <div className='content'>
                {currentTab.content}
            </div>
        )
    }
    
    render() {

        return(
            <div>
                {this.renderButtons()}
                {!!this.props.tabs.length && this.renderContent()}
            </div>
        )
    }
}