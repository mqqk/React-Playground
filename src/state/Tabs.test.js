import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Tabs from './Tabs';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Tab Component',() => {
    
    const tabsProp=[
        {
            name:'First Tab',
            content:"I'm just going to make something up here."
        },
        {
            name:'Second Tab',
            Content:"Here is something else to make up for Tab 2"
        },
        {
            name:'Third Tab',
            Content:"I hope the weather cools off soon."
        }
      ]


    it('renders without errors', () => {
        const div=document.createElement('div');
        ReactDOM.render(<Tabs />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders empty given no tabs',() =>{
        const wrapper = shallow(<Tabs />)
        expect(toJson(wrapper)).toMatchSnapshot()

    })

    it('renders the first tab by default', () =>{
        const wrapper = renderer.create(<Tabs tabs = {tabsProp} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('closes the first tab and opens any clicked tab', () => {
        const wrapper = shallow(<Tabs tabs={tabsProp} />)
        // console.log('>>>WRAPPER<<<')
        // console.log(wrapper.debug())
        // console.log('FIND (BUTTON)')
        // console.log(wrapper.find('button').at(1).debug())
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})

