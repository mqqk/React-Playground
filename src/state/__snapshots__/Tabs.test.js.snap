// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Component renders empty given no tabs 1`] = `<div />`;

exports[`Tab Component renders the first tab by default 1`] = `undefined`;
